apps/www/pages/_middleware.tsx

```diff
@@ -0,0 +1,52 @@
+import { NextFetchEvent, NextRequest, NextResponse } from 'next/server'
+import { parseAndVerifyJWT } from '../lib/auth/JWT/JWTHelper'
+import reportError from '../lib/errors/reportError'
+
+/**
+ * Middleware used to conditionally redirect between the marketing- and front-page
+ * depending on the user authentication status and roles.
+ * @param req
+ * @param ev
+ */
+export async function middleware(req: NextRequest, ev: NextFetchEvent) {
+  const resUrl = req.nextUrl.clone()
+
+  // Don't run the middleware unless on home-page
+  if (req.nextUrl.pathname !== '/') {
+    return NextResponse.next()
+  }
+
+  // Redirect to front-preview ssr to generate article front-preview
+  // used in the yearly overview
+  if (
+    !req.nextUrl.searchParams.has('marketing') &&
+    req.nextUrl.searchParams.has('extractId')
+  ) {
+    // Remap extractId query param to id-slug
+    const extractId = req.nextUrl.searchParams.get('extractId')
+    resUrl.searchParams.delete('extractId')
+
+    resUrl.pathname = `/_ssr/front-preview/${extractId}`
+
+    return NextResponse.rewrite(resUrl)
+  }
+
+  try {
+    // Parse and verify JWT to decide about redirection
+    const jwtBody = await parseAndVerifyJWT(req)
+    const isMember = jwtBody?.roles?.includes('member')
+
+    if (isMember) {
+      resUrl.pathname = '/front'
+      return NextResponse.rewrite(resUrl)
+    }
+  } catch (err) {
+    reportError(null, 'JWT Verification Error')
+    console.error(err)
```

Comment:
Sonst wird ja `reportError` geladen, was `useragent` lädt, was ja scheinbar ned geht ;)
```suggestion
    console.error('JWT Verification Error', err)
```

Reply:
Vögel!